lib/graphql_metrics/extractor.rb

```diff
@@ -133,17 +142,19 @@ def extract_batch_loaders
     def extract_query
       return unless respond_to?(:query_extracted)
 
-      start_time = ctx_namespace[START_TIME_KEY]
-      return unless start_time
+      start_time = ctx_namespace[START_TIME_KEY] if ctx_namespace
+      return unless start_time || ctx_namespace.nil?
 
       end_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)
 
+      duration = start_time && end_time ? end_time - start_time : nil
+
       query_extracted(
         {
           query_string: query.document.to_query_string,
           operation_type: query.selected_operation.operation_type,
           operation_name: query.selected_operation_name,
-          duration: end_time - start_time
+          duration: duration
```

Comment:
Change in contract here. There's no sensible duration for a query that didn't run.